src/map/atcommand.c

```diff
@@ -7253,12 +7253,16 @@ ACMD(mobinfo)
 			}
 #endif
 
-			if (item_data->slot)
-				snprintf(atcmd_output2, sizeof(atcmd_output2), " - %s[%d]  %02.02f%%", item_data->jname, item_data->slot, (float)droprate / 100);
-			else
-				snprintf(atcmd_output2, sizeof(atcmd_output2), " - %s  %02.02f%%", item_data->jname, (float)droprate / 100);
-
-			strcat(atcmd_output, atcmd_output2);
+			struct item link_item = { 0 };
+			link_item.nameid = monster->dropitem[i].nameid;
+			StringBuf buf;
+			StrBuf->Init(&buf);
+			StrBuf->AppendStr(&buf, " - ");
+			clif->format_itemlink(&buf, &link_item);
+			snprintf(atcmd_output2, sizeof(atcmd_output2), " %02.02f%%", (float)droprate / 100);
+			StrBuf->AppendStr(&buf, atcmd_output2);
+			strcat(atcmd_output, StrBuf->Value(&buf));
+			StrBuf->Destroy(&buf);
```

Comment:
This can be moved outside the for loop (both `Init()` and `Destroy()`), and here it could just `Clear()` it, to avoid unnecessary memory reallocations. See `ACMD(itemlist)` or `ACMD(who)` for examples.

If you want to further improve it, you can even remove the strcat() and use directly the StringBuf's `Value()` in `clif->message()` (again, examples of this in those same two functions I mentioned above)

Reply:
This is changed now, I think I made it better memory wise.
Removed atcmd_output2 entirely.
Initialized StrBuffer at item start and destroyed at item end.

Reworked the clear and write logic for the String buffer.

Also had forgot MVP drops so that is new too.